toolchain/check/handle_binding_pattern.cpp

```diff
@@ -10,10 +10,10 @@
 namespace Carbon::Check {
 
 auto HandleAddress(Context& context, Parse::AddressId parse_node) -> bool {
-  auto self_param_id =
-      context.node_stack().Pop<Parse::NodeKind::BindingPattern>();
-  auto self_param = context.insts().TryGetAs<SemIR::BindName>(self_param_id);
-  if (self_param && self_param->name_id == SemIR::NameId::SelfValue) {
+  auto self_param_id = context.node_stack().PopPattern();
```

Comment:
Maybe leave a TODO to restore some amount of type safety here when the infrastructure needed is available?

Reply:
I think this general `PopPattern` might actually be what we want here: according to the typed node, an `AddressPattern` can have `AnyPatternId` within it, and this function now diagnoses the case where it's applied to a pattern other than a binding pattern.